src/components/TDViewer/Action.jsx

```diff
@@ -36,24 +36,26 @@ export default function Action(props) {
     });
 
     const checkIfFormExists = (form) => {
-        for (const element of action.forms) {
-            for (const x of form.op) {
-                if (typeof (element.op) === 'string') {
-                    if (element.op === x.op) {
-                        return true;
-                    }
-                } else {
-                    if (element.op.includes(x)) {
-                        let deepCompare = true;
-                        for (const y in form) {
-                            if (y !== 'op') {
-                                if (element[y] !== form[y]) {
-                                    deepCompare = false;
+        if (action.forms) {
```

Comment:
Which block do you mean? 
the `if(action.forms)` checks if action has forms or if it is undefined, since it could be also a Thing Model without forms.

Reply:
Sorry, should have left a comment on the whole block instead of one line. 

I mean the entire `if (action.forms)` statement.

Reply:
refactored